### PR TITLE
Provide commit message format for Coverity defects

### PR DESCRIPTION
**Pull request by dinatale2**
### Description

Provide details about the commit message format for Coverity defect
fixes submitted. Will make changes to the commit message checker in the future.

### Motivation and Context

Document Coverity defect fix commit message expectations.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux code style requirements.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain `Signed-off-by`.
- [ ] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -223,3 +217,43 @@ Provide some porting notes here if necessary.
 OpenZFS-issue: https://www.illumos.org/issues/1234
 OpenZFS-commit: https://github.com/openzfs/openzfs/commit/abcd1234
 ```
+
+#### Coverity Defect Fixes
+If you are submitting a fix to a Coverity defect, the commit message must
```
Comment (behlendorf) [inline .github/CONTRIBUTING.md]: We should link to the Coverity results, https://scan.coverity.com/projects/zfsonlinux-zfs

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -223,3 +217,43 @@ Provide some porting notes here if necessary.
...
+meet the following guidelines:
+* The first line must be in the format of `Fix coverity defects: CID dddd, dddd...`
+where `dddd` represents each CID fixed by the commit.
+* The body of the commit message must be a list of Coverity defects and how
+each defect was corrected. Each defect must start in with a line in the format
```
Comment (behlendorf) [inline .github/CONTRIBUTING.md]: How about.

* The body of the commit message should list each Coverity defect and how it was corrected.

Comment (dinatale2) [reply .github/CONTRIBUTING.md]: Do you want me to leave the description of the format? Or should we just let the example fill in that part?

**Comment by dinatale2** [comment]
Small correction to the example commit message. Forgot the `CID` in the commit message subject line.

**Pull request merged**